src/main/docker/gitlab-jenkins-mysql.yml

```diff
@@ -0,0 +1,57 @@
+version: "2"
+services:
+    gitlab:
+        image: gitlab/gitlab-ce:latest
+        volumes:
+            - artemis-gitlab-data:/var/opt/gitlab
+            - artemis-gitlab-logs:/var/log/gitlab
+            - artemis-gitlab-config:/etc/gitlab
+        ports:
+            - "2222:22"
+            - "8081:80"
+        mem_limit: 3000m
+        hostname: gitlab
+        networks:
+            artemis:
+                ipv4_address: 172.33.0.2
+    jenkins:
+        build: jenkins
+        volumes:
+            - artemis-jenkins-data:/var/jenkins_home
+            - /var/run/docker.sock:/var/run/docker.sock
+            - /usr/local/bin/docker:/usr/bin/docker
+            - ./jenkins/jenkins-casc-config.yml:/usr/share/jenkins/ref/jenkins-casc-config.yml
+        ports:
+            - "8082:8080"
+        hostname: jenkins
+        environment:
+            - CASC_JENKINS_CONFIG=/usr/share/jenkins/ref/jenkins-casc-config.yml
+        networks:
+            artemis:
+                ipv4_address: 172.33.0.3
+    mysql:
+        command: mysqld --lower_case_table_names=1 --skip-ssl --character_set_server=utf8mb4 --explicit_defaults_for_timestamp
+        environment:
+            - MYSQL_USER=root
+            - MYSQL_ALLOW_EMPTY_PASSWORD=yes
+            - MYSQL_DATABASE=Artemis
+        image: mysql:8.0.23
+        ports:
+            - 3306:3306
+        volumes:
+            - ./data/.db:/var/lib/mysql
```

Comment:
Is there a reason besides backwards-compatibility why the MySQL database is mapped to a local directory and not a Docker volume like it is done with Gitlab and Jenkins data?

Reply:
Not having a `data` directory owned by root in the top level of the development directory seems like a good choice to me.
The only downside I can see is that users changing from a similar previous setup might have to start with a fresh database, copy the old one over into the volume, or manually change this entry back to the current state.

> Would it be better to change that line into - artemis-mysql-data:/var/lib/mysql?

I think so, yes. That change and adding `artemis-mysql-data` to the list of volumes at the bottom of the file should be enough to make it work.